### PR TITLE
Sandboxfs 0.1.0

### PR DESCRIPTION
**Pull request by misuchiru03**

### CODE REVIEW

**Comment by xtraeme** [comment]
#12481

**Comment by xtraeme** [comment]
Please rebase and make a single commit.

File: srcpkgs/sandboxfs/template [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'sandboxfs'
+pkgname=sandboxfs
+version=0.1.0
+revision=1
+wrksrc="sandboxfs-sandboxfs-${version}"
+archs="noarch"
```
Comment (xtraeme) [inline srcpkgs/sandboxfs/template]: really? I'd think this is wrong.

Comment (misuchiru03) [reply srcpkgs/sandboxfs/template]: The template is still being created incorrectly, as is addressed in another PR. I neglected to xlint the template this time.  Will fix.

Comment (misuchiru03) [reply srcpkgs/sandboxfs/template]: and removing the noarch statement

**Comment by abenson** [comment]
--- src/nodes/mod.rs.orig
    +++ src/nodes/mod.rs
    @@ -173,11 +173,11 @@
         }
         let size = size.unwrap();
     
    -    let result = if size > ::std::i64::MAX as u64 {
    -        warn!("truncate request got size {}, which is too large (exceeds i64's MAX)", size);
    +    let result = if size > ::nix::libc::off_t::max_value() as u64 {
    +        warn!("truncate request got size {}, which is too large (exceeds off_t's MAX)", size);
             Err(nix::Error::invalid_argument())
         } else {
    -        try_path(path, |p| unistd::truncate(p, size as i64))
    +        try_path(path, |p| unistd::truncate(p, size as nix::libc::off_t))
         };
         if result.is_ok() {
             attr.size = size;
    --- src/nodes/conv.rs.orig
    +++ src/nodes/conv.rs
    @@ -69,7 +69,7 @@
         } else {
             val.tv_usec() as i32
         };
    -    Timespec::new(val.tv_sec() as sys::time::time_t, usec)
    +    Timespec::new((val.tv_sec() as sys::time::time_t).into(), usec)
     }
     
     /// Converts a file type as returned by the file system to a FUSE file type.

**Comment by xtraeme** [comment]
@abenson please report this to upstream!

**Comment by abenson** [comment]
I do plan to! I don't know if the patch is correct, but it does solve an existing open issue!

**Pull request merged**